### PR TITLE
Small API refactoring

### PR DESCRIPTION
**Pull request by owlishDeveloper**
- Replaced duplicate code in my other APIs with external functions.
- Moved the functions to the top for readability.
- Improved the functions' descriptions a bit

### CODE REVIEW

File: src/api.js [hunk 1]
```diff
@@ -112,6 +112,38 @@ function resolve(res, status, message) {
   return res.status(status).send(message);
 }
 
+/***
+ Helper function to look up repo owner in the Azure table to get installation ID,
+ and authenticate with GitHub using that ID.
+
+ Receives owner's name, the Azure table, and github object.
+ Returns either authenticated github object or null
+***/
```
Comment (djmitche) [inline src/api.js:121]: I love the comment!

File: src/api.js [hunk 2]
```diff
@@ -112,6 +112,38 @@ function resolve(res, status, message) {
...
+async function installationAuthenticate(owner, directory, github) {
```
Comment (djmitche) [inline src/api.js]: Let's call the table `OwnersDirectory`, like it's called elsewhere.  That makes it easier to find with a string search.

File: src/api.js [hunk 3]
```diff
@@ -112,6 +112,38 @@ function resolve(res, status, message) {
...
+}
+
+/***
+ Helper function to find the most fresh status set by our bot.
+ Gets the bot's ID, gets statuses for the repo/branch, finds there the status by the bot's ID
+ 
+ Receives authenticated github object; names of owner, repo and branch; and configuration object
+ Returns either status object or undefined (if not found).
+***/
+async function findTCstatus(github, owner, repo, branch, configuration) {
```
Comment (djmitche) [inline src/api.js]: This is super-minor, but the first `s` in `status` should be capitalized (`Status`)

File: src/api.js [hunk 4]
```diff
@@ -305,14 +334,11 @@ api.declare({
     },
   };
 
-  if (ownerInfo) {
-    try {
-      let instGithub = await this.github.getInstallationGithub(ownerInfo.installationId);
+  let instGithub = await installationAuthenticate(owner, this.OwnersDirectory, this.github);
```
Comment (djmitche) [inline src/api.js:337]: This is a really nice refactor

**Review by djmitche — APPROVED** [review]
Good with some minor renames!

File: src/api.js (hunk 3 above)
Comment (owlishDeveloper) [reply src/api.js]: It's not capitalized to make "TC" stand out :) Still want to capitalize?

File: src/api.js (hunk 2 above)
Comment (owlishDeveloper) [reply src/api.js]: Done!

File: src/api.js (hunk 3 above)
Comment (djmitche) [reply src/api.js]: Yes - the first letter of each word is capitalized.

Comment (owlishDeveloper) [reply src/api.js]: Done!

**Pull request merged**